### ISSUE TITLE
device.getIpAddress is not a function error on settings page

### ISSUE THREAD

**Issue by torstengram** [robertklep/name.klep.sonoff#16]
Hello,
i am getting this error when trying to enter the settings page of the app
device.getIpAddress is not a function
it does not show me the 3 Tasmota devices i have, they seems to function OK

**Comment by robertklep** [comment]
Strange, that function should be defined for all Tasmota devices.

Could you try and restart the app to see if that solves the problem?

**Comment by torstengram** [comment]
Hi, restarting the app and the broker app (running on Homey) does not help. However right after restart i do not see the error message, but i do not see the devices either.
Anything i can extract in order to troubleshoot better? a log or something?

**Comment by robertklep** [comment]
There's a log, but it's not accessible unless you run the app from the command line.

Right after a restart it's expected that the devices aren't showing up, because it may take some time before the devices are detected.

What types of devices do you have?

**Comment by torstengram** [comment]
I have 2 PCs of S26 added as generic Tasmota device. And one POW

Regards

**Comment by robertklep** [comment]
Ah right, I think it might be because of those generic devices.

Try removing those and re-adding them as S20 devices (make sure you set the correct module type in the Tasmota configuration).

Be aware that removing devices will invalidate any flows that are built with them, though.

**Comment by thucar** [comment]
I'm having the same issue. I did add my S20, Bridge and Basic as generic devices because the S20 (newer version) and basic were not being picked up when I tried adding them as their actual devices.

The basic was not being detected at all when doing a soft restart during inclusion and when adding the S20, immediately after restart a message popped up saying "No new devices detected"

**Comment by thucar** [comment]
Tasmota firmware was configured correctly. I'm running the latest version: 6.3.0 with stock settings. Only configured WiFi, module type and MQTT  settings after uploading the firmware.

The S20 is definitely S20 but it looks like a newer version (S20_EU_V1.3) Not sure if it makes a difference.

When I tried adding it as a regular S20, I got the following error: 
![image](https://user-images.githubusercontent.com/2293862/48066477-65162080-e1d6-11e8-9b69-e7c329da43bd.png)


When trying to add the Basic, as I said, it never got detected at all. The only way for me to get thos two added was to use the Generic option. Then they were detected correctly and I managed to select their corresponding types.

**Comment by robertklep** [comment]
It looks like there's been a change in the Tasmota firmware regarding module identifiers. It'll take a few days before I can start fixing it; in the mean time, you could try and flash an earlier version of Tasmota.

**Comment by thucar** [comment]
It's ok. I was hoping I could define/edit sensor information on the settings page but looks like that's not the case. Am I correct to assume that currently only the Sonoffs stock capabilities get carried over when importing them to Homey? Meaning that if I have a Sonoff Basic with 12 DS12B20 sensors hooked up to it, I will not be able to access the temperature values from Homey?

I even looked into the MQTT Client app to see if I can access the data directly from the Broker. Unfortunately I was unable to break apart the tele/sonoff/SENSOR message into individual temperature readings from there either.

**Comment by robertklep** [comment]
> Am I correct to assume that currently only the Sonoffs stock capabilities get carried over when importing them to Homey? Meaning that if I have a Sonoff Basic with 12 DS12B20 sensors hooked up to it, I will not be able to access the temperature values from Homey?

Yes, that's the case right now. I will hopefully release a new version of the Homey app within the next two weeks or so that will be able to determine which sensors a device has, and map those sensors to (Homey) capabilities.
